mediasms_test.go

```diff
@@ -7,9 +7,35 @@ import (
 )
 
 func TestReplaceMessageBodyURLs(t *testing.T) {
-	messageBody := "http://google.com to https://github.com to http://stackoverflow.com and http://google.com"
-	allURLs := []string{"http://google.com", "https://github.com", "http://stackoverflow.com", "http://google.com"}
+	messageBodyA := "http://google.com to https://github.com to http://stackoverflow.com and http://google.com/mail"
+	allURLsA := []string{"http://google.com", "https://github.com", "http://stackoverflow.com", "http://google.com/mail"}
 
-	newMessageBody := ReplaceMessageBodyURLs(messageBody, allURLs)
-	assert.Equal(t, "{URL} to {URL2} to {URL3} and {URL4}", newMessageBody, "they should be equal")
+	newMessageBodyA := ReplaceMessageBodyURLs(messageBodyA, allURLsA)
+	assert.Equal(t, "{URL} to {URL2} to {URL3} and {URL4}", newMessageBodyA, "they should be equal")
+
+	messageBodyB := "google.com google.com/mail digima.com digima.com/らりるれる"
+	allURLsB := []string{"google.com", "google.com/mail", "digima.com", "digima.com/らりるれる"}
+
+	newMessageBodyB := ReplaceMessageBodyURLs(messageBodyB, allURLsB)
+	assert.Equal(t, "{URL} {URL2} {URL3} {URL4}", newMessageBodyB, "they should be equal")
+
+	messageBodyC := "http://google.com to https://github.com to http://stackoverflow.com and http://google.com/mail http://google.com"
+	allURLsC := []string{"http://google.com", "https://github.com", "http://stackoverflow.com", "http://google.com/mail"}
+
+	newMessageBodyC := ReplaceMessageBodyURLs(messageBodyC, allURLsC)
+	assert.Equal(t, "{URL} to {URL2} to {URL3} and {URL4} http://google.com", newMessageBodyC, "they should be equal")
```

Comment:
This test cases seems strange, shouldn't "http://google.com" be replaced with "{URL}"?

Reply:
We are only replacing the first link currently. Do you want to replace all of them? 

The purpose of this test case is the one you brought up where there are more links in the message then there are in the links table. I'm not sure how that could happen honestly.

If we wish to replace all of them, then instances where two or more links are identical will only have one replacement URL. e.g. `google.com google.com digima.com` -> `{URL} {URL} {URL2}`

Is this the desired behavior?

Reply:
We're confirming this at the moment.

Reply:
@CJButton @rdehaas-comvex We got the confirmation from @kazuhiro-comvex 
The URL should be the same so as the shortened. In that case, `google.com google.com digima.com` => `{URL} {URL} {URL2}` and regardless the order.